src/Draco.RedGreenTree.SourceGeneration/IncrementalGeneratorInitializationContextExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Draco.RedGreenTree.SourceGeneration;
+
+internal static class IncrementalGeneratorInitializationContextExtensions
+{
+    public static IncrementalValuesProvider<AttributeAndType> ForTypesWithAttribute(this IncrementalGeneratorInitializationContext ctx, string attributeName) =>
```

Comment:
Is the naming of this function good? If we want to fit into the LINQ-style, shouldn't we call it something like `WhereTypeHasAttribute`, or even just write an extension predicate and use `.Where(x => x.HasAttribute(...))`?

My SGs are quite rusty so there might be a limitation I forgot about.

Reply:
Kinda wanted to stick to the naming style of `ForAttributeWithMetadataName`, but sure.